### DISCUSSION TITLE
Sidebar integration in the pages

### DISCUSSION THREAD

**Discussion by Harikarthyk** [websight-io/starter#113] [Q&A] [answered]
Hi,

Can the sidebar be included in the components?

I checked the layouts, and there is no option for a sidebar.

I need something like a header and footer; is that possible to achieve?

**Sample Sidebar.**

![Sample](https://user-images.githubusercontent.com/54505967/210939107-32990ea8-eed1-4a96-9ffd-f84b57972cbd.png)

**Comment by paweljozwicki** [comment, marked as answer]
Hi, we don't have a sidebar in our components, and it's possible to achieve this, but only by preparing a dedicated component. There is no way to create one with the components we already have.

**Reply by Harikarthyk** [reply]
Okay, thank you.


It will be really helpful if I get any sample code for implementing them.

**Reply by paweljozwicki** [reply]
We can prepare something, but the question is if you use Howlite, Bulma, or your own components?

**Reply by Harikarthyk** [reply]
We are using our own components.

**Reply by Harikarthyk** [reply]
Hey
If possible, you can create a page with a header and sidebar using howlite, we will try to implement the same using our components. 

Thanks in advance.

**Reply by paweljozwicki** [reply]
I assume that in this case sidebar should just be a static part of the page so it's a matter of the page structure. To deal with our grid system I added three dedicated containers, and prepared a new page template:

<img width="336" alt="Screenshot 2023-01-16 at 07 58 20" src="https://user-images.githubusercontent.com/109656376/212617232-28ea3fcb-6608-4fa9-851f-716786c61edf.png">

offsetcontainer.html
``` 
<div class="hl-grid hl-cols-9">
  <sly data-sly-repeat.paragraph="${resource.children}" data-sly-resource="${paragraph.path @ resourceType=paragraph.resourceType}"></sly>
</div> 
```

sidebarcontainer.html
``` 
<div class="hl-grid hl-cols-3">
  <sly data-sly-repeat.paragraph="${resource.children}" data-sly-resource="${paragraph.path @ resourceType=paragraph.resourceType}"></sly>
</div> 
```

sidebarpagecontainer.html
``` 
<div class="hl-grid hl-cols-12">
  <sly data-sly-repeat.paragraph="${resource.children}" data-sly-resource="${paragraph.path @ resourceType=paragraph.resourceType}"></sly>
</div> 
```

and templates/sidebarpage/initial/.content.json
``` 
{
  "jcr:primaryType": "ws:Page",
  "jcr:content": {
    "jcr:primaryType": "ws:PageContent",
    "sling:resourceType": "howlite/components/page",
    "rootcontainer": {
      "sling:resourceType": "howlite/components/maincontainer/sidebarpagecontainer",
      "sidebarcontainer": {
        "sling:resourceType": "howlite/components/maincontainer/sidebarcontainer",
        "title": {"sling:resourceType": "howlite/components/container"}
      },
      "maincontainer": {
        "sling:resourceType": "howlite/components/maincontainer/offsetcontainer",
        "header": {"sling:resourceType": "howlite/components/headerreference"},
        "pagesection": {"sling:resourceType": "howlite/components/pagesection"},
        "footer": {"sling:resourceType": "howlite/components/footerreference"}
      }
    }
  }
}
```

And the result page layout:
<img width="1396" alt="Screenshot 2023-01-16 at 08 06 10" src="https://user-images.githubusercontent.com/109656376/212617808-ce5c21d1-db38-4555-8590-f5345a644fd8.png">

**Reply by rohitgupta274509** [reply]
Thanks for your suggestions, we also tried the same, using different containers we are able to get the required layouts.
